examples/regex-example.ts

```diff
@@ -40,85 +42,56 @@ const charmax10 = {
   message: 'Value must be 10 or fewer characters',
 }
 
-const FIELDS = {
-  supplierID: anmax50,
-  MBI: anmax11,
-  MemberZip: numexact5,
-  NPI: anmax15,
-  MemberAddress1: allcharmax50,
-  MemberAddress2: anmax50,
-  MemberAlias: anmax50,
-  MemberCity: lettersmax50,
-  Ending_DOS: charmax10,
-}
-
-const logger = { info: console.log }
-
-const fieldTypeVal = (record) => {
-  Object.keys(FIELDS).forEach((field) => {
-    const fieldRegex = FIELDS[field]['regex']
-    logger.info(FIELDS[field]['message'])
-    logger.info(fieldRegex.toString())
-    const fieldValue = record.get(field)
-    if (fieldRegex.test(fieldValue) === false) {
-      record.addError(field, FIELDS[field]['message'])
-      logger.info(FIELDS[field]['message'])
+const regexValidate = (regexMessage) => {
+  const retFunc = (val:string) => {
+    const regex = regexMessage.regex
+    const message = regexMessage.message
+    if (regex.test(val) === false) {
+      logger.info(regex.toString())
+      logger.info(message)
+      return [
+        new Message(message, 'error', 'validate')
+          ]
     }
-  })
```

Comment:
This recordCompute function is very hard to understand because it's doing a bunch of metaprogramming.  The whole platform-sdk framework is setup to do metaprogramming, use it.